### PR TITLE
14 key commands

### LINKED ISSUE #14: Key commands for interacting with modules
Allow the use to make use of the keyboard to control the configurator

### LINKED ISSUE #36: Fix rotate speed and button render.
Buttons shouldn't render if modules cant be rotated. Rotate speed should increased!

### PR DESCRIPTION
**Pull request by ConorTighe1995**
Users can now use the 'r' and 'delete' keys. The key 'r' allows the user to rotate modules clockwise and 'delete' key allows them to remove the module from the grid. The context menu will no longer renders the rotate buttons if the element can not be rotated. The rotate animation speed has been increased to improve usability.

closes #14
closes #36

### CODE REVIEW

**Comment by DanielSeehausen** [comment]
love the animations

File: src/components/Configurator.tsx [hunk 1]
```diff
@@ -434,8 +457,12 @@ class Configurator extends React.Component<any, ConfiguratorState> {
...
           >
             <Popup.Content>
-              <Button size='large' color='blue' icon='undo' circular onClick={this.rotateRight} />
-              <Button size='large' color='blue' icon='redo' circular onClick={this.rotateLeft} />
+              {isRotatable ? (
+                <>
+                  <Button size='large' color='blue' icon='undo' circular onClick={this.rotateRight} />
+                  <Button size='large' color='blue' icon='redo' circular onClick={this.rotateLeft} />
+                </>
+              ) : null}
```
Comment (vdiezel) [inline src/components/Configurator.tsx]: short circuiting is often a bit more readable: 

`isRotatable && ...`

File: src/lib/Module.ts [hunk 2]
```diff
@@ -51,16 +51,20 @@ export default class Module {
...
-      const transform = Module.GetTokenizedTransformAttributeValue(rotateClass.attr('transform'))
+    return rotateClass.nodes().length > 0
+  }
+
+  rotate(width: number, height: number, direction: boolean): void {
+    if (this.isRotatable()) {
+      const transform = Module.GetTokenizedTransformAttributeValue(d3.select(this.svg).select('.rotate').attr('transform'))
       const angle = transform.rotate ? parseInt(transform.rotate[0], 10) : 0
       if (angle % 90 === 0) {
         const rotateAnimation = () => d3.interpolateString(`rotate(${angle},${width / 2},${height / 2})`, `rotate(${angle + (direction ? 90 : -90)},${width / 2},${height / 2})`)
```
Comment (vdiezel) [inline src/lib/Module.ts]: some serious math here

File: src/lib/Module.ts [hunk 3]
```diff
@@ -51,16 +51,20 @@ export default class Module {
...
-  rotate(width: number, height: number, direction: boolean): void {
+  isRotatable() {
     const rotateClass = d3.select(this.svg).select('.rotate')
-    if (rotateClass.nodes().length > 0) {
-      const transform = Module.GetTokenizedTransformAttributeValue(rotateClass.attr('transform'))
+    return rotateClass.nodes().length > 0
+  }
+
+  rotate(width: number, height: number, direction: boolean): void {
+    if (this.isRotatable()) {
```
Comment (vdiezel) [inline src/lib/Module.ts]: invert the condition and return early:

```
if (!this.Rotatable) return 
// happy path
```

makes it more readable and easier to reason about

File: src/lib/Module.ts (hunk 2 above)
Comment (vdiezel) [inline src/lib/Module.ts]: this line is suspiciously long . what does the linter say? :grin:

File: src/lib/Module.ts (hunk 3 above)
Comment (ConorTighe1995) [reply src/lib/Module.ts]: Great point! ill update these later before merging!

File: src/lib/Module.ts (hunk 2 above)
Comment (ConorTighe1995) [reply src/lib/Module.ts]: I didn't write that so I didn't want to mess witth it too much but I might try move it into functions to make life easier down the line.

**Pull request merged**